### PR TITLE
fix: set 1000 sats as minimum amount that can be withdrawn

### PR DESCRIPTION
**Pull request by ws4charlie**
# Description

This PR is to prevent Bitcoin dust-amount withdrawals from registration. It set `1000 satoshis` as the minimum amount that can be withdrawn from zEVM.

Closes: [2326](https://github.com/zeta-chain/node/issues/2326)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: x/crosschain/keeper/evm_hooks.go [hunk 1]
```diff
@@ -327,7 +328,7 @@ func ValidateZrc20WithdrawEvent(event *zrc20.ZRC20Withdrawal, chainID int64) err
 	// The event was parsed; that means the user has deposited tokens to the contract.
 
 	if chains.IsBitcoinChain(chainID) {
-		if event.Value.Cmp(big.NewInt(0)) <= 0 {
+		if event.Value.Cmp(big.NewInt(constant.BTCWithdrawalDustAmount)) < 0 {
 			return fmt.Errorf("ParseZRC20WithdrawalEvent: invalid amount %s", event.Value.String())
```
Comment (swift1337) [inline x/crosschain/keeper/evm_hooks.go:332]: Let's add `constant.BTCWithdrawalDustAmount` to the error message to make it more helpful

**Review by swift1337 — APPROVED** [review]

File: x/crosschain/keeper/evm_hooks.go (hunk 1 above)
Comment (ws4charlie) [reply x/crosschain/keeper/evm_hooks.go]: https://github.com/zeta-chain/node/pull/2362/commits/7b122ebee5b2067759cf95fd2594becd0cb7a576

File: x/crosschain/keeper/evm_hooks.go [hunk 2]
```diff
@@ -327,8 +328,9 @@ func ValidateZrc20WithdrawEvent(event *zrc20.ZRC20Withdrawal, chainID int64) err
 	// The event was parsed; that means the user has deposited tokens to the contract.
 
 	if chains.IsBitcoinChain(chainID) {
-		if event.Value.Cmp(big.NewInt(0)) <= 0 {
-			return fmt.Errorf("ParseZRC20WithdrawalEvent: invalid amount %s", event.Value.String())
+		if event.Value.Cmp(big.NewInt(constant.BTCWithdrawalDustAmount)) < 0 {
+			return fmt.Errorf("ParseZRC20WithdrawalEvent: withdraw amount %s is less than minimum amount %d",
```
Comment (kingpinXD) [inline x/crosschain/keeper/evm_hooks.go:332]: NIT: We can define an error type for this  error

**Review by kingpinXD — APPROVED** [review]

File: x/crosschain/keeper/evm_hooks_test.go [hunk 3]
```diff
@@ -169,9 +170,16 @@ func TestValidateZrc20WithdrawEvent(t *testing.T) {
 			*sample.GetValidZRC20WithdrawToBTC(t).Logs[3],
 		)
 		require.NoError(t, err)
-		btcMainNetWithdrawalEvent.Value = big.NewInt(0)
+
+		// 1000 satoshis is the minimum amount that can be withdrawn
```
Comment (kingpinXD) [inline x/crosschain/keeper/evm_hooks_test.go:174]: maybe we can set the test name to 
`unable to validate a btc withdrawal event with an invalid amount`

**Comment by kingpinXD** [comment]
LGTM, 
Is there a way to double check that the original TX is reverted?

File: x/crosschain/keeper/evm_hooks.go (hunk 2 above)
Comment (ws4charlie) [reply x/crosschain/keeper/evm_hooks.go]: https://github.com/zeta-chain/node/pull/2362/commits/28973eaa3086ba0cbfafcddf5ad78c09efa9386b

File: x/crosschain/keeper/evm_hooks_test.go (hunk 3 above)
Comment (ws4charlie) [reply x/crosschain/keeper/evm_hooks_test.go:174]: https://github.com/zeta-chain/node/pull/2362/commits/28973eaa3086ba0cbfafcddf5ad78c09efa9386b

**Comment by ws4charlie** [comment]
> LGTM, Is there a way to double check that the original TX is reverted?

Yeah. It was reverted in my local network. See 
image

**Review by lumtis — APPROVED** [review]

**Comment by ws4charlie** [comment]
> > > LGTM, Is there a way to double check that the original TX is reverted?
> > 
> > 
> > Yeah. It was reverted in my local network. See image
> 
> I mean, would it make sense to add an e2e test and check the balance? The user should not have any loss of funds I guess , since the PostTxProcessing hook reverts the entire tx?
> 
> If you think an e2e test is not necessary just checking it would be a good idea

Seems just checking is okay. Also the 'unsupported address' validation was tested against mainnet on v17 release, the zEVM 'withdraw' tx just failed.

**Pull request merged**